lib/nose/backend.rb

```diff
@@ -174,10 +174,10 @@ def result_conditions(conditions, results)
               Condition.new field, :'=', result[field.id]
             end
 
-            unless @range_field.nil?
+            unless @range_fields.to_a.empty?
               operator = conditions.each_value.find(&:range?).operator
-              result_condition << Condition.new(@range_field, operator,
-                                                result[@range_field.id])
+              result_condition << Condition.new(@range_fields, operator,
+                                                result[@range_fields.id])
             end
```

Comment:
Normally each pair of operator and field corresponds to a single `Condition`object. Here multiple predicates (more than one range condition) are in a single object. It also seems to require that both of these have the same operator.